### PR TITLE
Tuchfarber/ent 1307 patch

### PR DESCRIPTION
**Pull request by tuchfarber**
**Description:** Fixes the issue that #419 introduced, while still completing ENT-1307

1.2.4 attempted to fix the side effect in HandleConsentEnrollment by
removing the view and calling the code as a method inside the post
call. It was ignoring the redirect url it was being passed and
ran all requests through this new method instead. This broke coupon
redemption and redirects from courseware.

This fixes by only using the new method when the redirect URL is a
specific string, thus allowing all previous pathways to work as they did
before.

ENT-1307

**JIRA:** ENT-1307

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] Called `make static` for webpack bundling if any static content was updated.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

**Comment by tuchfarber** [comment]
Review should focus on the second commit as the first was just a cherry pick of the old code + conflict fixes (changelog and version #): https://github.com/edx/edx-enterprise/pull/425/commits/a8e749daa650631718d19827f569686f88a19efb. The issue was that I thought all success URLs were heading to heading to the same view (`HandleConsentEnrollment`), but they can be anything. This only catches the requests that would have been forwarded to that view previously instead of all of them. 

Missed in manual testing originally because my LMS reverted to using the PyPI version, not the local version and I didn't realize.

**Comment by tuchfarber** [comment]
@edx/business-enterprise-team This is a redo of #419. I've tested it on the business sandbox and it's still there in the `enterprise/sprint_t` branch if you want to verify anything.

File: enterprise/views.py [hunk 1]
```diff
@@ -364,6 +365,87 @@ def get_course_or_program_context(self, enterprise_customer, course_id=None, pro
...
+        enterprise customer's course enrollment record should be created.
+
+        Depending on the value of query parameter `course_mode` then learner
+        will be either redirected to LMS dashboard for audit modes or
+        redirected to ecommerce basket flow for payment of premium modes.
+        """
+        # Redirect the learner to LMS dashboard in case no course mode is
+        # provided as query parameter `course_mode`
+        if not enrollment_course_mode:
+            return redirect(LMS_DASHBOARD_URL)
```
[the remark this answers is not in the document]
Comment (tuchfarber) [reply enterprise/views.py]: These were both pulled directly from the `HandleConsentEnrollment` view, so it's currently the same flow as what we have today. I'm happy to update it to a 404, I wasn't sure if there was historical reasoning for the dashboard redirect.

**Comment by tuchfarber** [comment]
@brittneyexline Do you need any additional information?

**Comment by christopappas** [comment]
Noticed this in my PRs and wanted to know if we are still planning to merge this in at all

**Comment by tuchfarber** [comment]
Ha! I totally forgot about this. Looks like I was waiting on more information. Also looks like it would need to be rebase a bit. Tickets still out there, so it should probably be picked back up. I can look at rebasing sometime next week. Mind if I ping you for review post-rebase?

**Comment by christopappas** [comment]
Yeah no prob at all @tuchfarber 👍

**Comment by christopappas** [comment]
Chatted about this offline. Gonna close for now, as we won't be revisiting this for a while.

**Pull request closed without merging**